<launch>

    <!-- Input data from RealSense camera -->
    <param name="topic_color_image" value="/camera/color/image_raw" /> 
    <param name="topic_camera_info" value="/camera/color/camera_info" /> 
    <param name="topic_depth_image" value="/camera/aligned_depth_to_color/image_raw" /> 
    <param name="file_camera_info" value="$(find auto_collect)/config/cam_params.json" />

    <!-- Files and communication settings -->
    <param name="folder_data" value="$(find auto_collect)/data/" /> 
    <param name="topic_point_cloud" value="/camera/depth/color/points" /> 
    <param name="topic_point_cloud_clustering_res" value="/my_cloud_clustering_res" /> 
    <param name="topic_point_cloud_objects" value="/my_cloud_objects" /> 
    <param name="topic_num_objects" value="/my_num_objects" /> 
    <param name="topic_objects_on_image" value="/my_objects_on_image" /> 


    <!-- Main nodes -->
    <node name="n0_fake_rgbd_image_publisher" pkg="auto_collect" type="n0_fake_rgbd_image_publisher.py"  output="screen"/>
    
    
    <node name="n1_main_add_label_to_image" pkg="auto_collect" type="n1_main_add_label_to_image.py"  output="screen">
        <param name="MAX_PROC_TIME_PER_FRAME" type="double" value="0.5" />     
    </node>


    <node name="node_detect_object_from_cloud" pkg="auto_collect"  type="node_detect_object_from_cloud" output = "screen">  
        <!-- segment plane -->
        <param name="num_planes" type="int" value="1" />     
        <param name="plane_distance_threshold" type="double" value="0.01" />     
        <param name="plane_max_iterations" type="int" value="100" />     

        <!-- divide cloud into clusters -->
        <param name="flag_do_clustering" type="bool" value="true" />     
        <param name="cluster_tolerance" type="double" value="0.02" />     
        <param name="min_cluster_size" type="int" value="50" />     
        <param name="max_cluster_size" type="int" value="10000" />   
        <param name="max_num_objects" type="int" value="3" />   
         
    </node>


    <!-- rviz -->
    <node type="rviz" name="rviz_1" pkg="rviz" args="-d $(find auto_collect)/config/rviz_rgb_and_color_image.rviz" respawn="true"/>
    <node type="rviz" name="rviz_2" pkg="rviz" args="-d $(find auto_collect)/config/rviz_cloud_src.rviz" respawn="true"/>
    <node type="rviz" name="rviz_3" pkg="rviz" args="-d $(find auto_collect)/config/rviz_cloud_res.rviz" respawn="true"/>

</launch>